client/src/partials/settings/settings.js

```diff
@@ -1,59 +1,62 @@
 angular.module('bhima.controllers')
-.controller('settings', [
-  '$scope',
-  '$routeParams',
-  '$translate',
-  '$location',
-  'appcache',
-  'messenger',
-  'tmhDynamicLocale',
-  'store',
-  'amMoment',
-  function($scope, $routeParams, $translate, $location, Appcache, messenger, tmhDynamicLocale, Store, amMoment) {
-    
-    // TODO issue discussing DB modelling of languages, quick suggestion (id, label, translateKey, localeKey)
-    var languageStore = new Store({identifier : 'translateKey'});
-    var languages = [
-      {
-        translateKey : 'en',
-        localeKey : 'en-us',
-        label : 'English'
-      },
-      {
-        translateKey : 'fr',
-        localeKey : 'fr-be',
-        label : 'French'
-      },
-      {
-        translateKey : 'ln',
-        localeKey : 'fr-cd',
-        label : 'Lingala'
-      }
-    ];
-    languageStore.setData(languages);
-    
-    $scope.url = $routeParams.url || '';
-    var cache = new Appcache('preferences');
+.controller('settings', Settings);
+
+Settings.$inject = ['$http','$routeParams','$translate','$location','appcache','messenger','tmhDynamicLocale','store','amMoment', 'SessionService'];
 
-    cache.fetch('language')
-    .then(function (res) {
-      if (res) {
-        $scope.settings = { language: res.translateKey };
-      }
-    });
+function Settings($http, $routeParams, $translate, $location, Appcache, messenger, tmhDynamicLocale, Store, amMoment, SessionService) {
+  var vm = this;
 
-    $scope.updateLanguage = function updateLanuage(key) {
-      var language = languageStore.get(key);
-      
-      $translate.use(language.translateKey);
-      tmhDynamicLocale.set(language.localeKey);
-      amMoment.changeLocale(language.translateKey);
+  // TODO issue discussing DB modelling of languages, quick suggestion (id, label, translateKey, localeKey)
```

Comment:
This comment was made very early on during the project however I think it
is still valid. I had addressed the modelling issue in a pull request that
never landed. A reliable service to cache and provide languages is a good
idea.

On Wednesday, January 27, 2016, Jonathan Niles <notifications@github.com>
wrote:

> In client/src/partials/settings/settings.js
> <https://github.com/IMA-WorldHealth/bhima-2.X/pull/42#discussion_r50953773>
> :
>
> >
> > -    $scope.updateLanguage = function updateLanuage(key) {
> > -      var language = languageStore.get(key);
> > -
> > -      $translate.use(language.translateKey);
> > -      tmhDynamicLocale.set(language.localeKey);
> > -      amMoment.changeLocale(language.translateKey);
> > +  // TODO issue discussing DB modelling of languages, quick suggestion (id, label, translateKey, localeKey)
>
> Also a good idea, if possible, include a client-side proposal for a
> LocaleService or LanguageService or the like.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/IMA-WorldHealth/bhima-2.X/pull/42/files#r50953773>.
>